coda/coda_replication/tests/test_views.py

```diff
@@ -0,0 +1,406 @@
+
+    def test_status_count_context_variable(self, client):
+        entries = factories.QueueEntryFactory.create_batch(30)
+
+        response = client.get(
+            reverse('coda_replication.views.queue_stats'))
+
+        status_counts = response.context[-1].get('status_counts')
+        assert len(status_counts) == 10
+
+        # Verify the calculated number of QueueEntries, grouped by status, are
+        # correct.
+        for status_id, status in STATUS_CHOICES:
+            assert status_counts.get(status) == sum(e.status == status_id for e in entries)
+
+    def test_template_used(self, client):
+        response = client.get(reverse('coda_replication.views.queue_stats'))
+        assert response.template[0].name == 'coda_replication/stats.html'
+
+
+class TestQueueSearch:
+    """
+    Tests for coda_replication.views.queue_search.
+    """
+
+    def test_status_ok_without_entries(self, rf):
+        request = rf.get('/')
+        response = views.queue_search(request)
+        assert response.status_code == 200
+
+    def test_status_ok(self, rf):
+        factories.QueueEntryFactory.create_batch(30)
+        request = rf.get('/')
+        response = views.queue_search(request)
+        assert response.status_code == 200
+
+    def test_entries_context_variable_is_page_object(self, client):
+        factories.QueueEntryFactory.create_batch(20)
+
+        response = client.get(
+            reverse('coda_replication.views.queue_search'))
+
+        page = response.context[-1]['entries']
+        assert isinstance(page, Page)
+
+    def test_blank_search_returns_all_queues(self, client):
+        entries = factories.QueueEntryFactory.create_batch(20)
+
+        response = client.get(
+            reverse('coda_replication.views.queue_search'))
+
+        results = response.context[-1]['entries']
+        assert len(entries) == len(results.object_list)
+
+    def test_filtering_by_status(self, client):
+        entries = factories.QueueEntryFactory.create_batch(20)
+        status = '2'
+
+        response = client.get(
+            reverse('coda_replication.views.queue_search'),
+            {'status': status})
+
+        results = response.context[-1]['entries']
+        assert sum(q.status == status for q in entries) == len(results.object_list)
+
+    def test_sorting_by_size(self, client):
+        """
+        Tests that the QueueEntry search results are ordered by the bytes
+        attribute in ascending order with the sort query is `size`.
+        """
+        entries = factories.QueueEntryFactory.create_batch(20)
+        response = client.get(
+            reverse('coda_replication.views.queue_search'),
+            {'sort': 'size'})
+
+        results = response.context[-1]['entries']
+        entries.sort(key=lambda x: x.bytes)
+
+        for entry, result in zip(entries, results.object_list):
+            assert entry.bytes == result.bytes
+
+    def test_filtering_by_harvest_end(self, client):
```

Comment:
`end_date` is not actually not an instance of `QueueEntry`, rather it is a datetime object. Instead of actually testing the result of the filter, this really verifies if the 'end_date` branch of the view is executed. The only time `context['end_date`]` is set is if that branch is executed. 

I don't love it, but it is better than checking if all of the models in the db >= a datetime object are present in the response context. :-1: 

To tell you the truth, I don't really like the idea of testing any of the sorting/filtering on the _search_ views.